### PR TITLE
Fix omitted snapshot-option

### PR DESCRIPTION
**Pull request by m-bucher**
Solves #269

### CODE REVIEW

File: lib/fog/vsphere/requests/compute/vm_take_snapshot.rb [hunk 1]
```diff
@@ -5,13 +5,13 @@ class Real
...
+            quiesce: false
+          }
           vm = get_vm_ref(options['instance_uuid'])
-          task = vm.CreateSnapshot_Task(
-            name: options['name'],
-            description: options['description'] || '',
-            memory: options['memory'] || true,
-            quiesce: options['quiesce'] || false
-          )
+          task = vm.CreateSnapshot_Task(defaults.merge(options))
```
Comment (ezr-ondrej) [inline lib/fog/vsphere/requests/compute/vm_take_snapshot.rb:14]: this won't work if I'm correct, options have (or can have at least) string keys.
```suggestion
          task = vm.CreateSnapshot_Task(defaults.merge(options.transform_keys(&:to_sym)))
```
This requires ruby `>= 2.5` but that should not be an issue I believe :thinking:

Comment (ezr-ondrej) [reply lib/fog/vsphere/requests/compute/vm_take_snapshot.rb:14]: or just have defaults in strings and continue expecting options will have string keys :)

Comment (m-bucher) [reply lib/fog/vsphere/requests/compute/vm_take_snapshot.rb]: The gemspec wants ruby `>=2.0.0`

File: lib/fog/vsphere/requests/compute/vm_take_snapshot.rb [hunk 2]
```diff
@@ -5,13 +5,17 @@ class Real
...
           raise ArgumentError, 'name is a required parameter' unless options.key? 'name'
+          defaults = {
+            description: '',
+            memory: true,
+            quiesce: false
+          }
+          opts = options.clone
+          defaults.each do |k, v|
+            opts[k] = v unless opts.key?(k) || opts.key?(k.to_s)
+          end
```
Comment (m-bucher) [inline lib/fog/vsphere/requests/compute/vm_take_snapshot.rb:16]: Not as smooth, but works with ruby-2.0 as well :wink:

**Comment by ezr-ondrej** [comment]
@chris1984 could you take it for a spin?

**Review by ezr-ondrej — APPROVED** [review]
LGTM :+1:

**Comment by chris1984** [comment]
Yeah will do

**Review by chris1984 — APPROVED** [review]
ACK, works good and no regressions with Foreman

**Pull request merged**